File: go/longestCommonPrefix.go
import (
	"fmt"
	"sort"
)

func longestCommonPrefix(strs []string) string {
	var ret []rune
	// i := 0
	var lens []int
	//
	for _, str := range strs {
		lens = append(lens, len(str))
	}

	sort.Ints(lens)
	if len(lens) == 0 {
		return ""
	}

	min_len := lens[0]
	// fmt.Printf("min_len: %v", min_len)

	for i := 0; i < min_len; i++ {
		var current_rune rune
		isCommon := true
		for idx, str := range strs {
			// fmt.Printf("i: %i, idx: %v, str: %v\n", i, idx, str)
			if idx == 0 {
				current_rune = []rune(str)[i]
			} else {
				if current_rune != []rune(str)[i] {
					isCommon = false
				} else {
					fmt.Printf("same")
				}
			}
		}
		if isCommon {
			ret = append(ret, current_rune)
		} else {
			break
		}
	}

	return string(ret)
}

File: interviews/python/longestCommonPrefix.py
from typing import List


class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs:
            return ""
        res = ""
        min_len = min(len(s) for s in strs)
        for idx in range(min_len):
            base = strs[0][idx]
            for s in strs[1:]:
                if base != s[idx]:
                    return res

            res += base

        return res
